File: clients/apps/lib/utils/relative_time.dart
String formatLastActive(String? iso) {
  if (iso == null || iso.isEmpty) return 'Offline';
  final t = DateTime.tryParse(iso);
  if (t == null) return 'Offline';

  final diff = DateTime.now().toUtc().difference(t.toUtc());
  if (diff.isNegative || diff.inSeconds < 60) return 'Active just now';

  final mins = diff.inMinutes;
  if (mins < 60) return 'Active ${mins}m ago';

  final hrs = diff.inHours;
  if (hrs < 24) return 'Active ${hrs}h ago';

  return 'Offline';
}

File: clients/packages/shared/src/utils/relativeTime.ts
export const formatLastActive = (iso?: string): string => {
  if (!iso) return 'Offline'
  const t = new Date(iso).getTime()
  if (isNaN(t)) return 'Offline'

  const diffMs = Date.now() - t
  if (diffMs < 0) return 'Active just now'

  const sec = Math.floor(diffMs / 1000)
  if (sec < 60) return 'Active just now'

  const min = Math.floor(sec / 60)
  if (min < 60) return `Active ${min}m ago`

  const hr = Math.floor(min / 60)
  if (hr < 24) return `Active ${hr}h ago`

  return 'Offline'
}
